src/simsopt/geo/surfacerzfourier.py

```diff
@@ -190,7 +194,10 @@ def from_wout(cls, filename: str, s: float = 1.0,
             interp = interp1d(s_full_grid, zmnc, kind=interp_kind, axis=0)
             zbc = interp(s)
 
-        mpol = int(np.max(xm))
+        # VMEC only considers poloidal modes up to `m=(mpol-1)`,
+        # but `SurfaceRZFourier` includes `m=mpol`.
+        # Here, `xm` comes from VMEC and `mpol` goes into `SurfaceRZFourier`.
+        mpol = int(np.max(xm)) - 1
```

Comment:
I don't think this -1 should be here. mpol here is SurfaceRZFourier's mpol, so it should equal the maximum m.